src/fiskaltrust.Launcher/ServiceInstallation/LinuxSystemD.cs

```diff
@@ -1,16 +1,29 @@
-﻿using fiskaltrust.Launcher.Helpers;
+﻿using fiskaltrust.Launcher.Common.Configuration;
+using fiskaltrust.Launcher.Helpers;
 using Serilog;
 
 namespace fiskaltrust.Launcher.ServiceInstallation
 {
     public class LinuxSystemD : ServiceInstaller
     {
         private static readonly string _servicePath = "/etc/systemd/system/";
-        private readonly string _serviceName = "fiskaltrustLauncher";
-
-        public LinuxSystemD(string? serviceName, LauncherExecutablePath launcherExecutablePath) : base(launcherExecutablePath)
+        private const string DefaultServiceName = "fiskaltrustLauncher";
+        private readonly string _serviceName;
+        private readonly string _serviceUser;
+        private readonly string _requiredDirectory;
+        
+        public LinuxSystemD(string? serviceName, LauncherExecutablePath launcherExecutablePath, LauncherConfiguration configuration)
+            : base(launcherExecutablePath)
         {
-            _serviceName = serviceName ?? _serviceName;
+            _serviceName = serviceName ?? DefaultServiceName;
+            _serviceUser = Environment.GetEnvironmentVariable("USER");
+
+            if (string.IsNullOrEmpty(_serviceUser))
+            {
+                Log.Warning("Service user name is not set. Owner of the service directory will not be changed.");
+            }
+
+            _requiredDirectory = configuration.ServiceFolder ?? throw new ArgumentNullException(nameof(configuration.ServiceFolder), "Service directory path must be provided in configuration.");
```

Comment:
```suggestion
            _requiredDirectory = configuration.ServiceFolder!;
```

We can just to it like that since this can not actually be null because of the default value in the configuration class...
I have not yet found a way though to tell that to the compiler so we don't have the null warnings 🥲